### PR TITLE
add cluster resource models docs

### PR DESCRIPTION
**Pull request by Poor12**
Signed-off-by: Poor12 <shentiecheng@huawei.com>

**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Add cluster resource models docs, refer to https://github.com/karmada-io/karmada/issues/2379.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: docs/userguide/scheduling/cluster-resource-models.md [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+      memory: 16265856Ki
+      pods: "110"
+    allocated:
+      cpu: 950m
+      memory: 290Mi
+      pods: "11"
+```
+
+However, the `ResourceSummary` is not precise enough, it mechanically counts the resources on all nodes, but ignores the fragment resources. For example, a cluster with 2000 nodes, 1 core CPU left on each node.
+From the `ResourceSummary`, we get that there are 2000 core CPU left for the cluster, but actually this cluster cannot run pods that require 2000 core CPU.
```
Comment (RainbowMango) [inline docs/userguide/scheduling/cluster-resource-models.md:28]:
```suggestion
From the `ResourceSummary`, we get that there are 2000 core CPU left for the cluster, 
but actually, this cluster cannot run any pod that requires CPU greater than 1 core.
```
Unusually, a pod requires 2000 core.

File: docs/userguide/scheduling/cluster-resource-models.md [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+
+Therefore, we introduce a `resource models` for each cluster which records the resource portraint of each node. Karmada will collect node and pod information for each cluster. After calculation, this node will be divided into the appropriate resource model configured by the users.
```
Comment (RainbowMango) [inline docs/userguide/scheduling/cluster-resource-models.md:30]:
```suggestion
Therefore, we introduce a `resource models` for each cluster that records the resource portrait of each node. Karmada will collect node and pod information for each cluster. After calculation, this node will be divided into the appropriate resource model configured by the users.
```

File: docs/userguide/scheduling/cluster-resource-models.md [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+        - --kubeconfig=/etc/kubeconfig
+        - --bind-address=0.0.0.0
+        - --cluster-status-update-frequency=10s
+        - --secure-port=10357
+        - --feature-gates=PropagateDeps=true,Failover=true,GracefulEviction=true,CustomizedClusterResourceModeling=true
+        - --v=4
+
+```
+
+After that, When a cluster is connected to the Karmada control plane, Karmada will automatically sets up a generic model for the cluster. You can see it in `cluster.spec`.
```
Comment (RainbowMango) [inline docs/userguide/scheduling/cluster-resource-models.md:54]:
```suggestion
After that, When a cluster is registered to the Karmada control plane, Karmada will automatically set up a generic model for the cluster. You can see it in `cluster.spec`.
```

Comment (Poor12) [inline docs/userguide/scheduling/cluster-resource-models.md:54]: when a cluster

File: docs/userguide/scheduling/cluster-resource-models.md [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+...
+...
+status:
+  - count: 1
+    grade: 7   
+```
+
+Assume that the Pod's request is 3C 20Gi. All nodes that meet Grade 2 and above meet this requirement. Considering the amount of available resources, the scheduler prefers to schedule the Pod to member3.
+
+| Cluster                   | member1                               | member2         | member3               |
```
Comment (Poor12) [inline docs/userguide/scheduling/cluster-resource-models.md:268]: need typo

File: docs/userguide/scheduling/cluster-resource-models.md [hunk 5]
```diff
@@ -0,0 +1,277 @@
...
+      min: "32"
+      name: cpu
+    - max: 512Gi
+      min: 256Gi
+      name: memory
+...
+...
+status:
+  - count: 1
+    grade: 7
```
Comment (Poor12) [inline docs/userguide/scheduling/cluster-resource-models.md:263]: grade：6

**Comment by RainbowMango** [comment]
Title: Cluster Resource Modeling

- Overview (describe the feature in general)
- Cluster Resource Modeling
  - General Cluster Modeling
  - Customized Cluster Modeling
- Disable Cluster Resource Modeling

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**